primitives/runtime/src/testing.rs

```diff
@@ -427,19 +426,20 @@ impl<Origin, Call, Extra, Info> Applyable for TestXt<Call, Extra> where
 
 	/// Executes all necessary logic needed prior to dispatch and deconstructs into function call,
 	/// index and sender.
-	fn apply<U: ValidateUnsigned<Call=Self::Call>>(
+	fn apply<U: ValidateUnsigned<Call = Self::Call>, D: crate::traits::Dispatcher<Self::Call>>(
 		self,
 		info: Self::DispatchInfo,
 		len: usize,
 	) -> ApplyExtrinsicResult {
-		let maybe_who = if let Some((who, extra)) = self.signature {
-			Extra::pre_dispatch(extra, &who, &self.call, info, len)?;
-			Some(who)
+		let (maybe_who, pre) = if let Some((who, extra)) = self.signature {
```

Comment:
Can we merge that with `fn apply` for `CheckedExtrinsic`?

Reply:
I de-duplicated this by factoring out `apply` to its own function. Alternative would be a blanket impl but I think that would have been even more ugly. Maybe you can have a look.